### PR TITLE
ARROW-11870: [Dev] Automatically run merge script in virtual environment

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@jorgecarleitao Could you try this out? It should be usable instead of the Python script (same CLI syntax).

Perhaps someone can also contribute a Windows (.bat) version.

File: dev/merge_arrow_pr.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        echo "Failed to setup virtual environment"
+        echo "Please delete directory '$ENV_DIR' and try again"
+        exit $?
+    }
+}
+
+[ -d $ENV_DIR ] || create_venv
+check_venv
+
+$ENV_PYTHON $GIT_ROOT/dev/merge_arrow_pr.py $@
```
Comment (kou) [inline dev/merge_arrow_pr.sh:56]:
```suggestion
$ENV_PYTHON $GIT_ROOT/dev/merge_arrow_pr.py "$@"
```

We need `"..."` to handle spaces correctly in shell.

```shell
#!/bin/sh

echo '$@'
python3 -c 'import sys; print(sys.argv)' $@
echo '"$@"'
python3 -c 'import sys; print(sys.argv)' "$@"
```

```console
$ /tmp/a.sh 1 "2 3" 4
$@
['-c', '1', '2', '3', '4']
"$@"
['-c', '1', '2 3', '4']
```

Comment (pitrou) [reply dev/merge_arrow_pr.sh:56]: Will do.

**Comment by pitrou** [comment]
@alamb Can you check whether this works for you?

**Review by jorgecarleitao — APPROVED** [review]
Sorry, @pitrou , this slipped my mind. I have tested this in a fresh clone and worked out of the box. LGTM!

I left a PR to your repo's branch with an update to the README.

**Comment by alamb** [comment]
@pitrou  I will give it a try later today. Thank you

**Comment by alamb** [comment]
I ran this locally on my mac and it seemed to work just fine (though I didn't actually merge a PR with it). 👍 

```
alamb@ip-10-0-0-124 arrow2 % ./dev/merge_arrow_pr.sh 

Creating Python virtual environment in /Users/alamb/Software/arrow2/dev/.venv3.9 ...

ARROW_HOME = /Users/alamb/Software/arrow2/dev
PROJECT_NAME = arrow
Which pull request would you like to merge? (e.g. 34): ^CTraceback (most recent call last):
  File "/Users/alamb/Software/arrow2/dev/merge_arrow_pr.py", line 597, in <module>
    cli()
  File "/Users/alamb/Software/arrow2/dev/merge_arrow_pr.py", line 559, in cli
    pr_num = get_pr_num()
  File "/Users/alamb/Software/arrow2/dev/merge_arrow_pr.py", line 547, in get_pr_num
    return input("Which pull request would you like to merge? (e.g. 34): ")
KeyboardInterrupt

alamb@ip-10-0-0-124 arrow2 % 
```

**Comment by pitrou** [comment]
Thanks for the feedback. I'll merge now.

**Pull request closed without merging**